### PR TITLE
Add more RX tests

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: tests/sim/PeripheralUART.cpp [hunk 1]
```diff
@@ -132,6 +132,26 @@ void PeripheralUART::ReceiveByte(USART_MODULE_ID index, uint8_t byte) {
...
+  // The logic here is a bit confusing, it looks like the parity and framing
+  // error bits are buffered along with the data byte.
+  if (uart.rx_enable) {
+    if (uart.rx_buffer.empty()) {
+      uart.errors |= USART_ERROR_FRAMING;
+      uart.rx_buffer.push(byte);
+      m_interrupt_controller->RaiseInterrupt(
+        static_cast<INT_SOURCE>(uart.interrupt_source));
+    } else {
+      uart.rx_buffer.push(0x8000 | byte);
```
Comment (peternewman) [inline tests/sim/PeripheralUART.cpp]: What's 0x8000, can we have a constant please?

File: tests/sim/SignalGenerator.cpp [hunk 2]
```diff
@@ -136,14 +141,17 @@ void SignalGenerator::ProcessNextEvent() {
...
-  // printf("clock %lld, event type %d %d %d\n", m_simulator->Clock(),
-  //         event.type, event.duration, event.byte);
   switch (event.type) {
     case EVENT_DELAY:
       AddDurationToClock(event.duration);
       m_state = WAITING;
       break;
     case EVENT_BREAK:
+      if (m_line_state == HIGH &&
+          (9 * m_cycles_per_bit) / m_cycles_per_microsecond < event.duration) {
```
Comment (peternewman) [inline tests/sim/SignalGenerator.cpp]: Can we have some more brackets around the left hand side of the < please

**Comment by peternewman** [comment]
Generally LGTM.

This is really impressive and surely some of the most well tested uC code about.

File: tests/sim/PeripheralUART.cpp (hunk 1 above)
Comment (nomis52) [inline tests/sim/PeripheralUART.cpp]: Done

**Comment by peternewman** [comment]
LGTM

**Pull request merged**